Upgrade-Guide/upgrade_to_3.10.md

```diff
@@ -0,0 +1,94 @@
+
+> **ALERT**: If any of your volumes, in the trusted storage pool that is being upgraded, uses disperse or is a pure distributed volume, this procedure is **NOT** recommended, use the [Offline upgrade procedure](#offline-upgrade-procedure) instead.
+
+#### Repeat the following steps, on each server in the trusted storage pool, to upgrade the entire pool to 3.10 version:
+1. Stop all gluster services, either using the command below, or through other means,
+    ```bash
+    #killall glusterfs glusterfsd glusterd
+    ```
+
+2. Stop all applications that run on this server and access the volumes via gfapi (qemu, NFS-Ganesha, Samba, etc.)
+
+3. Install Gluster 3.10
+
+4. Ensure that version reflects 3.10.0 in the output of,
+    ```bash
+    #gluster --version
+    ```
+
+5. Start glusterd on the upgraded server
+    ```bash
+    #glusterd
+    ```
+
+6. Ensure that all gluster processes are online by checking the output of,
+    ```bash
+    #gluster volume status
+    ```
+
+7. Self-heal all gluster volumes by running
+    ```bash
+    #for i in `gluster volume list`; do gluster volume heal $i; done
+    ```
+
+8. Ensure that there is no heal backlog by running the below command for all volumes
+    ```bash
+    #gluster volume heal <volname> info
+    ```
+> NOTE: If there is a heal backlog, wait till the backlog is empty, or the backlog does not have any entries needing a sync to the just upgraded server, before proceeding to upgrade the next server in the pool
+
+9. Restart any gfapi based application stopped previously in step (2)
+
+### Offline upgrade procedure
+This procedure involves cluster downtime and during the upgrade window, clients are not allowed access to the volumes.
+
+#### Steps to perform an offline upgrade:
+1. On every server in the trusted storage pool, stop all gluster services, either using the command below, or through other means,
+    ```bash
+    #killall glusterfs glusterfsd glusterd
+    ```
+2. Stop all applications that access the volumes via gfapi (qemu, NFS-Ganesha, Samba, etc.), across all servers
+
+3. Install Gluster 3.10, on all servers
+
+4. Ensure that version reflects 3.10.0 in the output of the following command on all servers,
+    ```bash
+    #gluster --version
+    ```
+
+5. Start glusterd on all the upgraded servers
+    ```bash
+    #glusterd
+    ```
+6. Ensure that all gluster processes are online by checking the output of,
+    ```bash
+    #gluster volume status
+    ```
+
+7. Restart any gfapi based application stopped previously in step (2)
+
+### Post upgrade steps
+Perform the following steps post upgrading the entire trusted storage pool,
+- It is recommended to update the op-version of the cluster. Refer, to the [op-version](./op_version.md) section for further details
```

Comment:
The edit to the op-version guide, suggests ways of getting the op-version that the cluster can support, I think it is best to leave it to that mechanism than overload a value here, thoughts?

Also, the suggestion to add an op-version table that you made, seems like a better place. Hard coding something when we have the flexibility of checking for the value does not seem prudent.